sdcm/utils/common.py

```diff
@@ -357,11 +357,37 @@ def inner(*args, **kwargs):
                 return return_val
             return inner
 
+        results = []
         with ThreadPoolExecutor(max_workers=self.num_workers) as pool:
             LOGGER.debug("Executing in parallel: '{}' on {}".format(func.__name__, self.objects))
             if not self.disable_logging:
                 func = func_wrap(func)
-            return list(pool.map(func, self.objects, timeout=self.timeout))
+            futures = []
+
+            for obj in self.objects:
+                futures.append(pool.submit(func, obj))
+
+            for future in futures:
+                try:
+                    result = future.result(self.timeout)
+                    results.append(FutureResult(exc=None, result=result))
+                except Exception as ex:  # pylint disable=broad-except
+                    if ignore_exceptions:
+                        LOGGER.warning('Error happend during running %s:\n%s', func.__name__, ex)
+                        results.append(FutureResult(exc=ex, result=None))
+                        continue
+                    raise
+
+        return results
+
+
+from dataclasses import dataclass
```

Comment:
please don't go into using this just yet, it's a python3.7 feature only, also we are need this PR backported to branch-3.1 and branch-3.2

Reply:
yes, already removed it